open-craft/openedx-completion-aggregator
# Add tracking logs

**Description:**

This PR is the first step for https://github.com/openedx/openedx-aspects/issues/222

It is a rebase of https://github.com/open-craft/openedx-completion-aggregator/pull/173 plus:
* emitted event prefix is `openedx` instead of `edx`
* removed `openedx.completion_aggregator.completion.*` events -- only emitting `progress`
* edx-completion upgraded to edx-platform version

Progress events will be emitted indicating any changes to aggregated progress for these block types:
```
    openedx.completion_aggregator.progress.course
    openedx.completion_aggregator.progress.chapter
    openedx.completion_aggregator.progress.sequential
    openedx.completion_aggregator.progress.vertical
```

**JIRA:** [FAL-3766](https://tasks.opencraft.com/browse/FAL-3766) (private link)

**Dependencies:** dependencies on other outstanding PRs, issues, etc. 

**Merge deadline:** ASAP -- there's follow-up work to come soon.

**Installation instructions:**
1. Install this package in your LMS venv
1. Ensure completion is enabled : should be a waffle switch enabled for [completion.enable_completion_tracking](http://local.edly.io:8000/admin/waffle/switch/)
1. Create a course with one or two subsections, sections, blocks.

Testing instructions:

1. Watch the tracking.log file generated by your LMS for completion events, e.g `tail -f $(tutor config printroot)/data/lms/logs/tracking.log | grep completion_aggregator`
1. Enrol in your test course and complete some blocks.
   * You should see `openedx.completion_aggregator.progress.*` events coming through to mark the progress towards completion of the units (vertical), subsections (sequential), sections (chapter), and course.
   * When you've completed the children of these blocks, you should see corresponding openedx.completion_aggregator.completion.* events.


**Reviewers:**

**Merge checklist:**

**Post merge:**
      finished.
